src/transform/free_vars.rs

```diff
@@ -144,7 +144,7 @@ fn collect_free_vars(rt: &mut RichTerm, free_vars: &mut HashSet<Ident>) {
             *deps = Some(new_deps);
         }
         Term::Array(ts, _) => {
-            for t in ts {
+            for t in ts.get_mut().unwrap().iter_mut() {
```

Comment:
Same remark about panicking. You probably want `make_mut()`. By the way, this is what this very function does about the RC inside the RichTerm at the beginning: we use `SharedTerm::make_mut`, which is just `Rc::make_mut` under the hood, and not `get_mut().unwrap()`.